app/src/main/java/com/zeronfinity/cpfy/MainActivity.kt

```diff
@@ -1,11 +1,78 @@
 package com.zeronfinity.cpfy
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.net.HttpURLConnection
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import kotlin.coroutines.CoroutineContext
+
+class MainActivity: AppCompatActivity(), CoroutineScope {
+    lateinit var job: Job
+
+    override val coroutineContext: CoroutineContext
+        get() = job + Dispatchers.Main
+
+    val contestApiUrl = "https://clist.by/api/v1/json/contest/"
+    val apiUsername = "Zeronfinity"
+    val apiKey = BuildConfig.CLIST_API_KEY
+    val apiDateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+    val numberOfDaysForApiEndDate = 7
+
+    data class ContestData(val name: String, val duration: Int, val platformName: String, val startTime: String, val endTime: String, val url: String)
+    val contestList = ArrayList<ContestData>()
+
+    private suspend fun getHttpConnectionData(url: String) : String =
+        withContext(Dispatchers.IO) {
+            try {
+                val urlConnection = URL(url).openConnection() as HttpURLConnection
+                urlConnection.inputStream.bufferedReader().readText()
+            } catch (e: Exception) {
+                e.printStackTrace()
+                ""
```

Comment:
What's with the ""? 
Don't we want to show some meaningful messages?

Reply:
Here "" represents an empty string that is returned if an exception is caught

Reply:
Yes, but shouldn't you return something meaningful like 'this error occured' so that the end user actually understands what happened here? 
Just a suggestion.

Reply:
Yes. I have given a new commit which shows a small toast message to user in case of network failure.